staging/src/k8s.io/component-helpers/nodedeclaredfeatures/helper.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodedeclaredfeatures
+
+import (
+	"fmt"
+	"slices"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/version"
+)
+
+// Helper provides functions for discovering node features and inferring pod feature requirements.
+// It is stateful and holds the feature registry.
+type Helper struct {
```

Comment:
Nit: I'd call this Registry or Framework. I think that's more consistent with other components following a similar design pattern.